### PR TITLE
Feature/genesis

### PR DESCRIPTION
**Pull request by imthatcarlos**
add /genesis endpoint and 3/4 methods + tests

Note: api/genesis/address/pages returns 404, so I commented the method out

### CODE REVIEW

File: lib/cardano_explorer/genesis.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    end
+
+    # NOTE: defined in the API docs, but returns 404
+    # https://cardanodocs.com/technical/explorer/api/#path--api-genesis-address-pages
+    #
+    # def pages(options = {})
+    #   valid_params = %w[page pageSize redeemeed]
+    #   options = options.keep_if { |k| valid_params.include?(k) }
+    #   get('address/pages', options).parsed_response['Right']
+    # end
```
Comment (philipqnguyen) [inline lib/cardano_explorer/genesis.rb]: So it turns out that their docs are out of date. The route to get the genesis addresses by a page number is now actually this:
`https://cardanoexplorer.com/api/genesis/address?page=1450`

There's no need to implement this now, in fact you could remove this comment, and I'll be happy to merge the rest (commented out code becomes messy later on). We can deal with querying genesis addresses later, unless you're very ambitious.

File: lib/cardano_explorer/genesis.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
+module CardanoExplorer
+  class Genesis < Base
+
+    def self.find
+      raise NoMethodError
+    end
+
+    def initialize
+    end
```
Comment (philipqnguyen) [inline lib/cardano_explorer/genesis.rb]: What do you think if in the `Base` class, we make the argument in `initialize` optional?
```ruby
# base.rb
def initialize(hash = nil)
  @hash = hash
end
```

Then we do not need to redefine initialize here.

File: lib/cardano_explorer/genesis.rb (hunk 1 above)
Comment (imthatcarlos) [reply lib/cardano_explorer/genesis.rb]: I see. Yeah okay, I've just removed the method for now. We can implement that later

File: lib/cardano_explorer/genesis.rb (hunk 2 above)
Comment (imthatcarlos) [reply lib/cardano_explorer/genesis.rb]: Yep, sounds good. Updated

**Comment by philipqnguyen** [comment]
released with 0.1.2

**Pull request merged**